tests/study_group/test_group_manager.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+from study_group.models import StudyGroup
+from django.contrib.auth.models import User
+
+
+@pytest.mark.django_db
+class TestStudyGroupManager:
+    def test_get_all_non_full_groups(self, persist_group, persist_second_group):
+        for i in range(persist_group.capacity):
+            persist_group.join_group(User.objects.create(username=f'user{i}', password='pass'))
+
+        result = StudyGroup.objects.get_all_non_full_groups()
+        assert persist_group not in result
+        assert persist_second_group in result
+
+    def test_search_group_by_keyword(self, persist_group, persist_second_group):
```

Comment:
Maybe we can use [parametrize](https://docs.pytest.org/en/6.2.x/parametrize.html) for doing this test?

Reply:
I agree, it would make more sense, I'll change it

Reply:
I couldn't figure a way to do something along the lines of:
`@pytest.mark.parametrize("keyword", "expected", [("this keyword shouldnt exist", []),
                                                     ("math", [persist_group]),
                                                     ("guitar", [persist_second_group]),
                                                     ("profile", [persist_group, persist_second_group])]
                             )`